src/Agent/NewRelic/Agent/Core/Logging/ThreadIdEnricher.cs

```diff
@@ -8,13 +8,24 @@
 
 namespace NewRelic.Agent.Core
 {
+    /// <summary>
+    /// Adds a tid property to the log event containing the current managed thread id
+    /// </summary>
     [NrExcludeFromCodeCoverage]
     internal class ThreadIdEnricher : ILogEventEnricher
     {
+        private LogEventProperty _tidProperty;
+
         public void Enrich(LogEvent logEvent, ILogEventPropertyFactory propertyFactory)
         {
-            logEvent.AddPropertyIfAbsent(propertyFactory.CreateProperty(
-                "tid", Thread.CurrentThread.ManagedThreadId));
+            var threadId = Thread.CurrentThread.ManagedThreadId;
+
+            var prop = _tidProperty;
+
+            if (prop == null || (int?)((ScalarValue)prop.Value).Value != threadId)
+                _tidProperty = prop = propertyFactory.CreateProperty("tid", threadId);
+
+            logEvent.AddPropertyIfAbsent(prop);
```

Comment:
Does this actually improve performance? Or is the simpler original code good enough? The only thing this code might save time with is creating a property. If doing that is relatively expensive you might be able to use a thread local cache for it in order to simplify the code.

Reply:
We can avoid unnecessary allocations by re-using the thread id property. The property is re-created anytime there's a change in the thread id, so maybe you're right that caching by thread id would be a better solution. I'll look into that.